tutorons/common/dblogger.py

```diff
@@ -0,0 +1,53 @@
+from tutorons.common.models import Block, ServerQuery
+from tutorons.common.htmltools import get_css_selector
+import datetime
+
+def extract_HTTP_request_fields(request):
+        url = request.POST.get('origin')
+        path = request.path_info
+        ip = request.META.get('HTTP_X_FORWARDED_FOR')
+        if ip:
+            ip = ip.split(',')[-1].strip()
+        else:
+            ip = request.META.get('REMOTE_ADDR')
+        return url, path, ip
+
+class DBLogger():
+
+    def log_query(self, request):
+        # extract request fields
+        url, path, ip = extract_HTTP_request_fields(request)
+        q = ServerQuery(ip_addr=ip, path=path)
+        q.save()
+        return q.id
+
+    def log_region(self, request, region):
+        url, path, ip = extract_HTTP_request_fields(request)
+        r_type = request.path_info.split('/')[1]
+        r_method = request.path_info.split('/')[2]
+        block_hash = hash(region.node)
+        block_type = region.node.name
+        # find corresponding query
+        q = ServerQuery.objects.filter(ip_addr=ip, path=path)[0]
```

Comment:
This isn't quite right.  The first `ServerQuery` entry from an IP to a path might not be the right one.  Probably, we should just pass in the ServerQuery that was just created.  May require some additional data passing for calls to `explain` methods, where explanations don't occur at the same time as the initial query.